### PR TITLE
Fixing bug: No Rename Check When Rename Input Is Empty

### PR DESCRIPTION
**Pull request by AinaMerch**
## Proposed Changes
Fixing bug related to previous pull request ([#7024 ](https://github.com/MarkUsProject/Markus/pull/7024)). The check is triggered even when no rename input is provided. This should not happen. Changed to the check triggering only if the input is not empty. Also added a test case for the same.

## Type of Change
*(Write an `X` or a brief description next to the type or types that best describe your changes.)*

| Type                                                                                    | Applies? |
|-----------------------------------------------------------------------------------------|----------|
| 🚨 *Breaking change* (fix or feature that would cause existing functionality to change) |          |
| ✨ *New feature* (non-breaking change that adds functionality)                          |          |
| 🐛 *Bug fix* (non-breaking change that fixes an issue)                                  |     X     |
| 🎨 *User interface change* (change to user interface; provide screenshots)              |          |
| ♻️ *Refactoring* (internal change to codebase, without changing functionality)          |          |
| 🚦 *Test update* (change that *only* adds or modifies tests)                            |          |
| 📦 *Dependency update* (change that updates a dependency)                               |          |
| 🔧 *Internal* (change that *only* affects developers or continuous integration)         |          |

## Checklist
*(Complete each of the following items for your pull request. Indicate that you have completed an item by changing the `[ ]` into a `[x]` in the raw text, or by clicking on the checkbox in the rendered description on GitHub.)*

Before opening your pull request:

- [X] I have performed a self-review of my changes.
  - Check that all changed files included in this pull request are intentional changes.
  - Check that all changes are relevant to the purpose of this pull request, as described above.
- [X] I have added tests for my changes, if applicable.
  - This is **required** for all bug fixes and new features.

After opening your pull request:

- [X] I have updated the project Changelog (this is required for all changes).
- [X] I have verified that the pre-commit.ci checks have passed.
- [X] I have verified that the CI tests have passed.
- [X] I have reviewed the test coverage changes reported by Coveralls.
- [X] I have [requested a review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) from a project maintainer.

## Questions and Comments
*(Include any questions or comments you have regarding your changes.)*

### CODE REVIEW

File: app/assets/javascripts/Components/__tests__/submission_file_upload_modal.test.jsx [hunk 1]
```diff
@@ -196,6 +196,16 @@ describe("For the SubmissionFileUploadModal", () => {
         "file2.txt"
       );
     });
+
+    it("should call onSubmit prop with correct arguments when rename input is left blank", () => {
```
Comment (david-yz-liu) [inline app/assets/javascripts/Components/__tests__/submission_file_upload_modal.test.jsx:200]: Great! Since you used `.trim` above (which is good), please add another test case for when the rename input consists only of whitespace.

**Review by david-yz-liu — APPROVED** [review]
